deepmd/pt/infer/deep_eval.py

```diff
@@ -39,19 +39,25 @@ def __init__(
         auto_batch_size: Union[bool, int, AutoBatchSize] = True,
     ):
         self.model_path = model_file
-        state_dict = torch.load(model_file, map_location=env.DEVICE)
-        if "model" in state_dict:
-            state_dict = state_dict["model"]
-        self.input_param = state_dict["_extra_state"]["model_params"]
-        self.input_param["resuming"] = True
-        self.multi_task = "model_dict" in self.input_param
-        assert not self.multi_task, "multitask mode currently not supported!"
-        self.type_split = self.input_param["descriptor"]["type"] in ["se_e2_a"]
-        self.type_map = self.input_param["type_map"]
-        self.dp = ModelWrapper(get_model(self.input_param, None).to(DEVICE))
-        self.dp.load_state_dict(state_dict)
-        self.rcut = self.dp.model["Default"].descriptor.get_rcut()
-        self.sec = np.cumsum(self.dp.model["Default"].descriptor.get_sel())
+        if str(self.model_path).endswith(".pt"):
+            state_dict = torch.load(model_file, map_location=env.DEVICE)
+            if "model" in state_dict:
+                state_dict = state_dict["model"]
+            self.input_param = state_dict["_extra_state"]["model_params"]
+            self.input_param["resuming"] = True
+            self.multi_task = "model_dict" in self.input_param
+            assert not self.multi_task, "multitask mode currently not supported!"
+            model = get_model(self.input_param, None).to(DEVICE)
+            model = torch.jit.script(model)
+            self.dp = ModelWrapper(model)
+            self.dp.load_state_dict(state_dict)
+        elif str(self.model_path).endswith(".pth"):
+            model = torch.jit.load(model_file, map_location=env.DEVICE)
+            self.dp = ModelWrapper(model)
+        else:
+            raise ValueError("Unknown model file format!")
+        self.rcut = self.dp.model["Default"].rcut
+        self.type_map = self.dp.model["Default"].type_map
```

Comment:
If #3247 got merged, I can do the change: `rcut` -> `get_rcut()`; `type_map` -> `get_type_map()`.

Reply:
https://github.com/deepmodeling/deepmd-kit/pull/3247 merged.

Reply:
```suggestion
        self.rcut = self.dp.model["Default"].get_rcut()
        self.type_map = self.dp.model["Default"].get_type_map()
```